File: src/my_py_pkg/my_py_pkg/reset_counter_client.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.srv import SetBool
from functools import partial

class ResetCounterClient(Node):
  def __init__(self):
    super().__init__("reset_counter_client")
    self.client_ = self.create_client(SetBool, "reset_counter")
    self.get_logger().info("ResetCounterClient has been started")

  def call_reset_counter(self, do_reset:bool):
    while not self.client_.wait_for_service(1.0):
      self.get_logger().warn("Waiting for ResetCounter server...")

    request = SetBool.Request()
    request.data = do_reset

    future = self.client_.call_async(request)

    future.add_done_callback(partial(self.call_reset_counter_callback, request=request))

  def call_reset_counter_callback(self, future, request:SetBool.Request):
    response = future.result()
    self.get_logger().info(
      response.message + " (" + str(request.data) + " -> " + str(response.success) + ")"
    )

def main(args=None):
  rclpy.init(args=args)
  node = ResetCounterClient()
  node.call_reset_counter(False)
  node.call_reset_counter(True)
  rclpy.spin(node)
  rclpy.shutdown()


if __name__ == "__main__":
  main()

File: src/my_cpp_pkg/src/reset_counter_client.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/set_bool.hpp"

using namespace std::chrono_literals;
using namespace std::placeholders;

class ResetCounterClient : public rclcpp::Node
{
public:
  ResetCounterClient() : Node("reset_counter_client")
  {
    client_ = this-> create_client<example_interfaces::srv::SetBool>("reset_counter");
    request_ = std::make_shared<example_interfaces::srv::SetBool::Request>();
    RCLCPP_INFO(this->get_logger(), "ResetCounterClient node started");
  }

  void callResetCounter(bool data)
  {
    while(!client_->wait_for_service(1.0s))
    {
      RCLCPP_WARN(this->get_logger(), "Waiting for the server...");
    }

    request_->data = data;
    client_->async_send_request(request_, std::bind(&ResetCounterClient::callResetCounterCallback, this, _1));
  }

private:
  void callResetCounterCallback(rclcpp::Client<example_interfaces::srv::SetBool>::SharedFuture future)
  {
    auto response = future.get();
    RCLCPP_INFO(this->get_logger(), "%s (%s -> %s)", response->message.data(), (request_->data) ? "true" : "false", (response->success) ? "true" : "false");
  }

  rclcpp::Client<example_interfaces::srv::SetBool>::SharedPtr client_;
  example_interfaces::srv::SetBool::Request::SharedPtr request_;
};

int main(int argc, char **argv)
{
  rclcpp::init(argc, argv);
  auto node = std::make_shared<ResetCounterClient>();
  node->callResetCounter(false);
  node->callResetCounter(true);
  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}